### PR TITLE
eigenpy 2.6.6 (new formula)

### PR DESCRIPTION
**Pull request by jcarpent**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by jcarpent** [comment]
@carlocab While the original project (https://github.com/stack-of-tasks/eigenpy) does not meet the stat of 30 followers, etc. it is used by many other libraries (e.g. https://github.com/stack-of-tasks/pinocchio) that I would like to add to brew and which do have the stats mentioned by the linter. Would it be possible to accept this formula even if the stats are not totally fulfilled?

File: Formula/eigenpy.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  depends_on "doxygen" => :build
+  depends_on "boost-python3"
+  depends_on "eigen"
+  depends_on "numpy"
+  depends_on "python@3.9"
+
+  def install
+    pyver = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
+    py_prefix = Formula["python@3.9"].opt_frameworks/"Python.framework/Versions/#{pyver}"
+    ENV.prepend_path "PYTHONPATH", Formula["numpy"].opt_lib/"python#{pyver}/site-packages"
```
Comment (carlocab) [inline Formula/eigenpy.rb:20]:
```suggestion
    pyver = Language::Python.major_minor_version "python3"
    py_prefix = Formula["python@3.9"].opt_frameworks/"Python.framework/Versions"/pyver
    ENV.prepend_path "PYTHONPATH", Formula["numpy"].opt_prefix/Language::Python.site_packages("python3")
```

File: Formula/eigenpy.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    ENV.prepend_path "Eigen3_DIR", Formula["eigen"].opt_share/"eigen3/cmake"
+
+    mkdir "build" do
+      args = *std_cmake_args
+      args << "-DPYTHON_EXECUTABLE=#{py_prefix}/bin/python#{pyver}"
```
Comment (carlocab) [inline Formula/eigenpy.rb:25]:
```suggestion
      args << "-DPYTHON_EXECUTABLE=python3
```
Doesn't this work?

**Comment by carlocab** [comment]
Looks like this meets the notability check now.

File: Formula/eigenpy.rb (hunk 2 above)
Comment (jcarpent) [reply Formula/eigenpy.rb:25]: This is not working. We need a full path.

**Review by jcarpent — COMMENTED** [review]
Thanks @carlocab for the complete review.

**Comment by jcarpent** [comment]
@carlocab Do you know how to handle the Linux failure?

File: Formula/eigenpy.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  def install
+    pyver = Language::Python.major_minor_version "python3"
+    py_prefix = Formula["python@3.9"].opt_bin
+    ENV.prepend_path "PYTHONPATH", Formula["numpy"].opt_prefix/Language::Python.site_packages("python3")
+    ENV.prepend_path "Eigen3_DIR", Formula["eigen"].opt_share/"eigen3/cmake"
+
+    mkdir "build" do
+      args = *std_cmake_args
+      args << "-DPYTHON_EXECUTABLE=#{py_prefix}/bin/python#{pyver}"
```
Comment (carlocab) [inline Formula/eigenpy.rb]:
```suggestion
    python = Formula["python@#{pyver}"].opt_bin/"python#{pyver}"
    ENV.prepend_path "PYTHONPATH", Formula["numpy"].opt_prefix/Language::Python.site_packages("python3")
    ENV.prepend_path "Eigen3_DIR", Formula["eigen"].opt_share/"eigen3/cmake"

    mkdir "build" do
      args = *std_cmake_args
      args << "-DPYTHON_EXECUTABLE=#{python}"
```
You've duplicated the `bin` directory here. Let's try to make this simpler.

File: Formula/eigenpy.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  def install
+    pyver = Language::Python.major_minor_version "python3"
+    py_prefix = Formula["python@3.9"].opt_prefix
+    ENV.prepend_path "PYTHONPATH", Formula["numpy"].opt_prefix/Language::Python.site_packages("python3")
+    ENV.prepend_path "Eigen3_DIR", Formula["eigen"].opt_share/"eigen3/cmake"
+
+    mkdir "build" do
+      args = *std_cmake_args
+      args << "-DPYTHON_EXECUTABLE=#{py_prefix}/bin/python#{pyver}"
```
Comment (carlocab) [inline Formula/eigenpy.rb:25]:
```suggestion
    python = Formula["python@#{pyver}"].opt_bin/"python#{pyver}"
    ENV.prepend_path "PYTHONPATH", Formula["numpy"].opt_prefix/Language::Python.site_packages("python3")
    ENV.prepend_path "Eigen3_DIR", Formula["eigen"].opt_share/"eigen3/cmake"

    mkdir "build" do
      args = *std_cmake_args
      args << "-DPYTHON_EXECUTABLE=#{python}"
```

**Review by carlocab — APPROVED** [review]
Thanks, @jcarpent.

**Pull request closed without merging**